### DISCUSSION TITLE
2 copies of react-dom.production.min.js with app router?

### DISCUSSION THREAD

**Discussion by CaptainN** [vercel/next.js#51296] [App Router] [answered]
I'm seeing 2 copies of react-dom.production.min.js in my bundles, when using bundle analyzer. At least I think that is what this means. Is there some way to trace why that is happening?

At first I thought it might be from using a mono-repos with pnpm workspaces, but even if I clean install node_modules in just the web folder with yarn or npm, I still get the same result. 

![image](https://github.com/vercel/next.js/assets/245825/1f980c33-5e20-4f46-93d3-b746553c1e22)

**Comment by icyJoseph** [comment, marked as answer]
One is for the pages router. It is still not fully removed, but it won't be served to the client if you don't have pages dir pages.

For example, if you don't have a global not-found in App Dir, then the Pages Dir 400 page kicks in, and the other modules are shipped to be able to do hydration. This has let to some people claiming the app dir has a larger bundle size, when it is in fact, pages dir + app dir. 

Check yourself, when you go to your app dir page, how much JS is transferred, and what files come down?

- https://github.com/Makers-Den/next-bundle-test/issues/1

I do believe they are working on, if not merged already to 13.4.5, a way to not generated pages stuff if there's no pages at all.

**Reply by CaptainN** [reply]
Cool, I'll keep an eye out for that change. Thanks!
